### DISCUSSION TITLE
Send mail on login throttle lock

### DISCUSSION THREAD

**Discussion by Julien-degardin** [symfony/symfony#49771] [Q&A]
Hi everyone !

Do you know if it is possible to send an email to a raw-coded address when someone is locked by the login throttling system? (even better, add the forcing IP address within the message?)
I'm working on Symfony 6.0
If you think it is doable, I'd love to know how I could achieve this !

Thanks!

**Comment by mfettig** [comment]
It may be. If I was looking at doing this, I would start with trying an Event Listener configured for "LoginFailureEvent"
https://symfony.com/doc/current/security.html#authentication-events

Once you have the event in your listener, you should be able to check if the exception is of type LockedException and perform your email logic there. LockedException should have the HttpFoundation request object, which should be able to get you the IP you're looking for

https://github.com/symfony/symfony/blob/6.0/src/Symfony/Component/Security/Http/Event/LoginFailureEvent.php
https://github.com/symfony/symfony/blob/6.0/src/Symfony/Component/Security/Core/Exception/LockedException.php
